core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -207,6 +207,10 @@ package object config {
     .booleanConf
     .createWithDefault(false)
 
+  private[spark] val APP_CALLER_CONTEXT = ConfigBuilder("spark.log.callerContext")
```

Comment:
My concern is that this is a very hadoop specific thing at the moment, and it is unclear other environments will support it, so it'd make more sense to have hadoop in the name. A lot of Spark users don't run Hadoop.